### PR TITLE
mlperf fixes

### LINKED ISSUE #750: [ml_perf]validate use train.flags
https://github.com/tensorflow/minigo/blob/master/ml_perf/reference_implementation.py#L190
train.flags should be used for training, but not validation?
```
def validate(state, holdout_glob):
  checked_run('validation',
      'python3', 'validate.py', holdout_glob,
      '--flagfile={}'.format(os.path.join(FLAGS.flags_dir, 'train.flags')),
      '--work_dir={}'.format(fsdb.working_dir()))
```

I should update the report as some flags in train.flags may not be used by validate.py.  Should put common part in a seperate file and include from both train.flags and validate.flags

### LINKED ISSUE #751: [ml_perf]eval failed bacause double pass
I met this error when running one of the eval games
```
[F] game.cc:88 : check failed: !game_over_                                   
                                                                             
  0x41a127  minigo::Game::SetGameOverBecauseOfPasses()                       
  0x417404  minigo::MctsPlayer::PlayMove()                                   
  0x409d0a  minigo::(anonymous namespace)::Evaluator::ThreadRun()            
  0x7f65f80f4ad0  execute_native_thread_routine                              
```
Some debugging shows that game_over condition had not been checked between two moves:
https://github.com/tensorflow/minigo/blob/master/cc/eval.cc#L205
```
      curr_player->PlayMove(move);
      next_player->PlayMove(move);
```
It should be changed to the following and the game can finish normally.
```
      curr_player->PlayMove(move);
      if (game.game_over()) {
        break;
      }
      next_player->PlayMove(move);
```

### LINKED ISSUE #752: [ml_perf] The way flags files are threated are not friendly to list parameters
When define
```
--lr_rates=0.016
--lr_rates=0.0016
--lr_rates=0.00016
--lr_boundaries=25000
--lr_boundaries=37500
```
Only the last item of same name will be reserved
In expand_flags (https://github.com/tensorflow/minigo/blob/master/ml_perf/reference_implementation.py#L96)
```
  for arg in expanded:
    flag = arg.split('=', 1)[0]
    deduped[flag] = arg
```
Should change to the following code
```
  for arg in expanded:
    flag = arg.split('=', 1)[0]
    if flag in deduped.keys():
      # in the case of --lr_rate and --lr_boundaries, same key might appear
      # multiple times, make sure they all appear in command arg list
      deduped[flag] += " {}".format(arg)
    else:
      deduped[flag] = arg
```
and .flags files should be managed in a way that flags that should not appear twice or more won't appear twice or more.

### PR DESCRIPTION
**Pull request by tommadams**
fixes #752
fixes #751
fixes #750 

also adds slow window support

### CODE REVIEW

File: ml_perf/flags/19/train.flags [hunk 1]
```diff
@@ -1,8 +1,6 @@
...
+# Flags for training.
+
+--flagfile=ml_perf/flags/19/architecture.flags
 
---conv_width=64
---fc_width=64
---trunk_layers=5
---value_cost_weight=0.125
 --filter_amount=0.1
 --shuffle_buffer_size=10000
```
Comment (amj) [inline ml_perf/flags/19/train.flags]: we have lots more train flags to add, but i'll put them in another PR.

**Comment by tommadams** [comment]
updated the slow window logic, ptal

**Comment by delock** [comment]
Two flags inherited from selfplay.flags in eval.flags still causing cc:eval to fail.  Do you also encounter same error?
```
--holdout_pct=0.03
--disable_resign_pct=0.1
```
```
$ OMP_NUM_THREADS=1 KMP_AFFINITY=granularity=fine,proclist=[0],explicit bazel-bin/cc/eval --engine=tf --parallel_games=4 --parallel_games=2 --num_rea
douts=160 --virtual_losses=8 --value_init_penalty=2 --holdout_pct=0.03 --disable_resign_pct=0.1 --resign_threshold=-0.99 --model=/home/gma/delock-minigo/results/mlt-skx054/models/00000
2-000002.pb --model_two=/home/gma/delock-minigo/results/mlt-skx054/models/000001-000001.pb --sgf_dir=/home/gma/delock-minigo/results/mlt-skx054/sgf/eval/000002-000002 --seed=3 --instan
ce_id=0
ERROR: unknown command line flag 'disable_resign_pct'
ERROR: unknown command line flag 'holdout_pct'
```

**Comment by tommadams** [comment]
No, we don't see this. As I said earlier, we don't pass the expanded flags on the command line. The expanded flags are just for logging.

You need to pass the flag file unchanged.

**Comment by tommadams** [comment]
Concretely: https://github.com/tensorflow/minigo/blob/7439171589e65eafa72aa61c929339e3d7611791/ml_perf/reference_implementation.py#L100

Notice how we pass the `cmd` arguments to `subprocess.run` and not `expanded`.

**Pull request merged**